datafusion-postgres/src/handlers.rs

```diff
@@ -752,18 +844,25 @@ impl Parser {
             )));
         }
 
-        // show statement may not be supported by datafusion
-        if sql_trimmed.starts_with("show") {
-            // Return a dummy plan for transaction commands - they'll be handled by transaction handler
-            let show_schema =
-                Arc::new(Schema::new(vec![Field::new("show", DataType::Utf8, false)]));
-            let df_schema = show_schema.to_dfschema()?;
-            return Ok(Some(LogicalPlan::EmptyRelation(
-                datafusion::logical_expr::EmptyRelation {
-                    produce_one_row: true,
-                    schema: Arc::new(df_schema),
-                },
-            )));
+        // Parse and check for SET/SHOW statements using structured AST
+        if let Ok(parsed_statements) = crate::sql::parse(sql) {
+            if let Some(statement) = parsed_statements.first() {
+                if matches!(
+                    statement,
+                    SqlStatement::SetVariable { .. } | SqlStatement::ShowVariable { .. }
```

Comment:
we should not include `SetVariable` here, it doesn't return a resultset like show